src/types/auth.ts

```diff
@@ -0,0 +1,27 @@
+import { User, Session } from "next-auth";
+import { AdapterUser } from "next-auth/adapters";
+
+export interface UserWithRole extends User {
+  profile: {
+    role: UserType;
+  };
+}
+
+export interface ExtendedSession extends Session {
+  user: Session["user"] & {
+    role: UserType;
+  };
+}
+
+export interface CustomAdapterUser extends AdapterUser {
+  profile?: {
+    role: UserType;
+  };
+}
+
+export enum UserType {
```

Comment:
nice!

Reply:
Honestly I'm wondering if it would be possible to just use the default TypeScript enum where UserType.User = 0, UserType.Para = 1, and then use it as our ROLE_LEVEL, and instead have some fixed function called getUserTypeString(userType: UserType) that returns "user", "para", "case_manager", "admin", and is used only for writing to the database, which stores these strings rather than the enum.